src/optimizer_interface.jl

```diff
@@ -27,12 +27,24 @@ end
 
 
 """
-    function optimize!(model::Model,
-                       optimizer_factory::Union{Nothing, OptimizerFactory}=nothing;
-                       ignore_optimize_hook=(model.optimize_hook === nothing))
+    optimize!(model::Model,
+              optimizer_factory::Union{Nothing, OptimizerFactory}=nothing;
+              ignore_optimize_hook=(model.optimize_hook === nothing))
 
-Optimize the model. If `optimizer_factory` is not `nothing`, it first set the
-optimizer to a new one created using the optimizer factory.
+Optimize the model. If `optimizer_factory` is not `nothing`, it first sets the
+optimizer to a new one created using the optimizer factory. The factory can be
+created using the [`with_optimizer`](@ref) function.
+
+## Examples
```

Comment:
I'd suggest two separate examples, one where the optimizer is provided with `Model()` and another when it's provided with `optimize!`.